warehouse/oidc/models/google.py

```diff
@@ -54,6 +56,21 @@ class GooglePublisherMixin:
 
     __unchecked_claims__ = {"azp", "google"}
 
+    @staticmethod
+    def __lookup_all__(klass, signed_claims: SignedClaims) -> Query | None:
+        return Query(klass).filter_by(
+            email=signed_claims["email"], sub=signed_claims["sub"]
+        )
+
+    @staticmethod
+    def __lookup_no_sub__(klass, signed_claims: SignedClaims) -> Query | None:
+        return Query(klass).filter_by(email=signed_claims["email"], sub=None)
+
+    __lookup_strategies__ = [
+        __lookup_all__,
+        __lookup_no_sub__,
+    ]
```

Comment:
So the idea here is that the `email` claim is required and the `sub` claim is optional? Makes sense to me I think.

Reply:
Yep, exactly 🙂